base/client.jl

```diff
@@ -66,19 +68,52 @@ function repl_cmd(cmd, out)
         ENV["OLDPWD"] = new_oldpwd
         println(out, pwd())
     else
-        @static if !Sys.iswindows()
+        local command
+        if Sys.iswindows()
+            if shell_name == "cmd"
+                command = _CMD_execute(shell, cmd)
+            elseif shell_name in ("powershell", "pwsh")
+                command = _powershell_execute(shell, cmd)
+            elseif shell_name == "busybox"
+                command = `$shell sh -c $(shell_escape_posixly(cmd))`
+            else
+                command = `$shell $cmd`
+            end
+        else
             if shell_name == "fish"
                 shell_escape_cmd = "begin; $(shell_escape_posixly(cmd)); and true; end"
+                command = `$shell -c $shell_escape_cmd`
+            elseif shell_name == "pwsh"
+                command = _powershell_execute(shell, cmd)
```

Comment:
Doesn't this do the wrong escaping on Unix?

Reply:
(I'm not sure it's worth the effort to support `pwsh` on Unix.   This is only used for interactive mode, after all.)

Reply:
Powershell's handling of arguments is known (https://github.com/PowerShell/PowerShell/issues/1995) to be buggy, so can't reliably be used for launching programs anyways.

Reply:
@vtjnash, that's argument handling on Windows.  Presumably on Unix systems `pwsh` just uses `*argv`, no?

Reply:
Sadly, no (e.g. https://github.com/PowerShell/PowerShell/issues/10440)

> no CommandLineToArgvW layer

Sadly, it is currently present (although both broken and undocumented—and eventually supposed to be unnecessary)